### PR TITLE
use pyupgrade for python3.7+

### PR DESCRIPTION
**Pull request by FranzForstmayr**
This is an autogenerated change from the pyupgrade tool (https://github.com/asottile/pyupgrade)
Due to the drop of python3.6 I thought we could update the codebase to a newer syntax standard. For example old string formatters are converted to f-strings and old python2 code is removed.

### CODE REVIEW

**Comment by Vinc0110** [comment]
Cool. I did not even know I'm not supposed to use `'a={}'.format(a)` anymore...

File: skrf/vi/scpi/parser.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import re
 import sys
 
-kwarg_pattern = re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \(\)\'\"]*)>')
+kwarg_pattern = re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \\(\\)\'\"]*)>')
```
Comment (jhillairet) [inline skrf/vi/scpi/parser.py:6]: is the result the same?

**Review by jhillairet — COMMENTED** [review]
While I totally agree that f-string is an improvement in readability, I'm not that a fan of replacing the `set` and `dict` function with just `{}`. I find it less clear when reading, especially to understand when it is a set and not a dict...

File: skrf/vi/scpi/parser.py (hunk 1 above)
Comment (FranzForstmayr) [reply skrf/vi/scpi/parser.py:6]: Hmm, i have to check

Comment (FranzForstmayr) [reply skrf/vi/scpi/parser.py:6]: Yes it is:

```python
import re
re1 = re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \(\)\'\"]*)>')
re2 = re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \\(\\)\'\"]*)>')
print(re1)
print(re2)
print(re1 == re2)
```

```
re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \\(\\)\'"]*)>')
re.compile('<([a-zA-Z0-9_]+=?[a-zA-Z0-9_, \\(\\)\'"]*)>')
True
```

**Comment by FranzForstmayr** [comment]
@jhillairet should  I return the `set`/`dict` related changes?

**Comment by jhillairet** [comment]
> @jhillairet should I return the `set`/`dict` related changes?

is it easy? If not, no big deal.

**Comment by FranzForstmayr** [comment]
I could do it manually of course, however I like the `dict` comprehension and find it more readable than a list comprehension of tuples. 
Your decision...

**Comment by jhillairet** [comment]
Ok leave it like the PR. I was more concerned with the set. Knowing it is a
set rather than a list is important..

**Comment by FranzForstmayr** [comment]
Ok, I'll check the sets

**Pull request merged**